Assets/Mirror/Runtime/Transport/ITransport.cs

```diff
@@ -10,14 +11,32 @@ public enum TransportEvent { Connected, Data, Disconnected }
     public interface ITransport
     {
         // client
-        bool ClientConnected();
+        event Action ClientConnected;
+        event Action<byte[]> ClientDataReceived;
+        event Action<Exception> ClientError;
+        event Action ClientDisconnected;
```

Comment:
why not keep existing function names and name those OnClientConnected etc.? would be consistent with OnServerAddPlayer and so on

Reply:
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members

> ✓ DO name events with a verb or a verb phrase.
> 
> Examples include Clicked, Painting, DroppedDown, and so on.
> 
> ✓ DO give events names with a concept of before and after, using the present and past tenses.
> 
> For example, a close event that is raised before a window is closed would be called Closing, and one that is raised after the window is closed would be called Closed.
> 
> X DO NOT use "Before" or "After" prefixes or postfixes to indicate pre- and post-events. Use present and past tenses as just described.
>